src/games/strategy/triplea/ui/AbstractForumPosterPanel.java

```diff
@@ -21,13 +25,23 @@
   protected IPlayerBridge m_bridge;
   protected PBEMMessagePoster m_poster;
   protected TripleAFrame m_frame;
-  protected Action m_doneAction;
   protected ForumPosterComponent m_forumPosterComponent;
+  protected AbstractAction m_doneAction = new AbstractAction("Done") {
+    private static final long serialVersionUID = -5084680807785728744L;
+
+    @Override
+    public void actionPerformed(final ActionEvent e) {
+      if (m_forumPosterComponent.getHasPostedTurnSummary() || JOptionPane.YES_OPTION ==
```

Comment:
The existing logic has changed. It's not clear why we have the new boolean conditional. What is the significance of `getHasPostedTurnSummary` relative to the JOptionPane prompt? That is not clear in the context of the code why the external dependency exists.

But you also have not answered this question:
> If we can avoid the getHasPostedTurnSummary() method call that would be good. Is it possible to initializes the m_doneAction in the constructor and know what this value will be without needing to get it from m_forumPosterComponent. Would you mind commenting on that and/or investigating?